src/pages/signin/SAMLSignInPage/index.js

```diff
@@ -0,0 +1,68 @@
+import React, {useEffect} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import compose from '../../../libs/compose';
+import ONYXKEYS from '../../../ONYXKEYS';
+import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
+import CONFIG from '../../../CONFIG';
+import Icon from '../../../components/Icon';
+import Text from '../../../components/Text';
+import * as Expensicons from '../../../components/Icon/Expensicons';
+import * as Illustrations from '../../../components/Icon/Illustrations';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+
+const propTypes = {
+    /** The credentials of the logged in person */
+    credentials: PropTypes.shape({
+        /** The email/phone the user logged in with */
+        login: PropTypes.string,
+    }),
+    ...withLocalizePropTypes,
+};
+const defaultProps = {
+    credentials: {},
+};
+
+function SAMLSignInPage(props) {
+    useEffect(() => {
+        window.open(`${CONFIG.EXPENSIFY.SAML_URL}?email=${props.credentials.login}&referer=${CONFIG.EXPENSIFY.EXPENSIFY_CASH_REFERER}`, '_self');
```

Comment:
It's possible that this page can be accessed on native when `newDotSAML` beta enabled

https://github.com/Expensify/App/blob/4308871ec475d6346406b30a1302765eb0d30793/src/pages/signin/SignInPage.js#L92-L96

Reply:
🤔 why would it crash? `credentials.login` is the only portion we'd need to be defined and we'll only show the `SAMLEnabledForm` (now `ChooseSSOOrMagicCode`) or intiaite the SAML login (i.e. navigate to that URL immediately) if the login is set.